proto/workflow.proto

```diff
@@ -110,6 +113,39 @@ message GetWorkflowsResponse {
   repeated Workflow workflow = 2;
 }
 
+message ExecuteWorkflowRequest {
+  // The request context.
+  context.RequestContext request_context = 1;
+
+  // ID of the workflow to be executed.
+  // Ex. "WF4576963743584254779"
+  string workflow_id = 2;
+
+  // Name of the workflow action to execute. Corresponds to an action in
+  // buildbuddy.yaml, or an action name in the default workflow config if
+  // buildbuddy.yaml is missing from the repo.
+  // Ex. "Test all targets"
+  string action_name = 3;
+
+  // SHA of the commit to be checked out in order to execute the workflow.
+  // Ex. "e782592faf56da05cc0a243220689135e807958f"
+  string commit_sha = 4;
+
+  // Branch corresponding to the SHA to be checked out.
+  // Ex. "cool-feature"
+  string branch = 5;
+}
+
+message ExecuteWorkflowResponse {
+  // The response context.
+  context.ResponseContext response_context = 1;
+
+  // The invocation ID of the workflow execution. This invocation will not exist
+  // in the DB until the CI runner executes and kicks off the invocation, so
+  // clients may need to poll the API to check for existence.
+  string invocation_id = 2;
```

Comment:
Feels wrong to return a resource that doesn't exist yet. How do you plan to use this?

Reply:
wdyt about siggi's idea in the other PR about creating a bare-bones invocation in the DB?

Reply:
The reason it works to do that in the invocation handler is because it's during the lifetime of a request, and once the request hangs up we can close out that invocation. Splitting it across requests breaks that model, so if workflows start failing we could end up with a bunch of fake invocations in the DB.

I would much rather mask this in the UI than in the database, because the cost is a lot lower.

Reply:
I think this is resolved now because we decided to return an invocation ID from the API after all -- PTAL